source/jormungandr/jormungandr/scenarios/helper_classes/helper_utils.py

```diff
@@ -155,118 +180,167 @@ def _extend_journey(pt_journey, fallback_dp, fallback_period_extremity):
+    """
+    fallback_logic = _get_fallback_logic(fallback_type)
+    pt_obj = fallback_logic.get_pt_boundaries(pt_journey)
+
+    if pt_obj.uri == requested_obj.uri:
+        # No need for a crowfly if the pt section starts from the requested object
+        return None
+
+    if pt_obj.uri in places_free_access.odt:
+        pt_obj.CopyFrom(requested_obj)
+        return None
+
+    section_datetime = fallback_logic.get_pt_section_datetime(pt_journey)
+    pt_datetime = fallback_logic.get_journey_bound_datetime(pt_journey)
+    fallback_duration = fallback_durations[pt_obj.uri].duration
+
+    crowfly_dt = fallback_logic.get_fallback_datetime(pt_datetime, fallback_duration)
+
+    crowfly_origin, crowfly_destination = fallback_logic.route_params(requested_obj, pt_obj)
+    begin, end = fallback_logic.route_params(crowfly_dt, section_datetime)
+
+    return _create_crowfly(pt_journey, crowfly_origin, crowfly_destination, begin, end, mode)
+
+
+def _build_fallback(
+    requested_obj,
+    pt_journey,
+    mode,
     streetnetwork_path_pool,
-    dest_accessible_by_crowfly,
-    dest_fallback_durations_pool,
+    obj_accessible_by_crowfly,
+    fallback_durations_pool,
     request,
+    fallback_type,
 ):
+    accessibles_by_crowfly = obj_accessible_by_crowfly.wait_and_get()
+    fallback_durations = fallback_durations_pool.wait_and_get(mode)
+    fallback_logic = _get_fallback_logic(fallback_type)
 
-    accessibles_by_crowfly = dest_accessible_by_crowfly.wait_and_get()
-    fallback_durations = dest_fallback_durations_pool.wait_and_get(arr_mode)
+    pt_obj = fallback_logic.get_pt_boundaries(pt_journey)
 
-    for pt_journey in pt_journeys.journeys:
-        pt_destination = pt_journey.sections[-1].destination
-        last_section_end = pt_journey.sections[-1].end_date_time
-
-        if requested_dest_obj.uri != pt_destination.uri:
-            fallback_period_extremity = PeriodExtremity(pt_journey.arrival_date_time, True)
+    if requested_obj.uri != pt_obj.uri:
+        if pt_obj.uri in accessibles_by_crowfly.odt:
+            pt_obj.CopyFrom(requested_obj)
+        else:
             # extend the journey with the fallback routing path
-            direct_path_type = StreetNetworkPathType.ENDING_FALLBACK
+            represent_start = fallback_logic.represent_start()
+            pt_datetime = fallback_logic.get_pt_section_datetime(pt_journey)
+            fallback_period_extremity = PeriodExtremity(pt_datetime, represent_start)
+            orig, dest = fallback_logic.route_params(requested_obj, pt_obj)
+
             fallback_dp = streetnetwork_path_pool.wait_and_get(
-                pt_destination,
-                requested_dest_obj,
-                arr_mode,
-                fallback_period_extremity,
-                direct_path_type,
-                request=request,
+                orig, dest, mode, fallback_period_extremity, fallback_type, request=request
```

Comment:
streetnetwork_path_pool has a future manager running inside of it, which is no longer running inside of a context...